### PR TITLE
Adds public process status field to the project summary tab

### PR DESCRIPTION
**Pull request by tillyw**
## Associated issues
fixes https://github.com/cityofaustin/atd-data-tech/issues/10833 and
fixes https://github.com/cityofaustin/atd-data-tech/issues/11096

this pr adds the new 'public process' field in the ui, along with the db enhancements @johnclary made in [a previous pr](https://github.com/cityofaustin/atd-moped/pull/924). rather than adding a new component, i refactored the existing lead and sponsor component to work for any field that uses a single-select dropdown menu, and fed in all unique values as props. hopefully the props are readable and this will make it easier for us to add new fields as we go!

## Testing
**URL to test:** 
https://10833_public_process_field_ui--atd-moped-main.netlify.app/moped/

**Steps to test:**
- go to a project summary page and note the blank 'public process' field
- add, edit and delete values for the field
- change the value on the field and click the x to reset to the previous value
- do the same for lead and entity fields

---
#### Ship list
- [x] Code reviewed 
- [x] Product manager approved
- [ ] Added to [QA test script](https://docs.google.com/spreadsheets/d/1n_O6MLh9cwwPf57HUM394Ea-z9uuoEV1-QW4axNZXLE/edit#) if applicable

### CODE REVIEW

**Review by roseeichelmann — APPROVED** [review]
This works like a charm and i really like the refactor you did, i think it was a great way of adding this field to project summary. Super clean, well done!

File: moped-database/metadata/tables.yaml
Comment (chiaberry) [inline moped-database/metadata/tables.yaml:4006]: im glad we are moving towards the only select permissions for lookup tables!

File: moped-editor/src/views/projects/projectView/ProjectSummary/ProjectSummaryAutocomplete.js
Comment (chiaberry) [inline moped-editor/src/views/projects/projectView/ProjectSummary/ProjectSummaryAutocomplete.js:12]: i was going to try to suggest a different name for this variable since I couldn't tell what the name column is, but I can't think of anything better.

**Review by chiaberry — APPROVED** [review]
ran through the test steps and its all good!

**Review by frankhereford — APPROVED** [review]
This looks great Tilly. Thanks again for coming up with the generalized-use component - it's a much more elegant solution. 

I see that some of my comments are concerning changes which were inherited from the pair of commits in PR 924 from @johnclary. All good, and those migration comments are low priority anyway -- they are more about developer readability in the future and not about if this will work or not.

I don't want to withhold approval though; I'll let you decide what is worth pursuing and what isn't. Let me know if I can clarify anything. Thanks!

File: moped-database/migrations/1672774236734_create_table_public_moped_public_process_status/up.sql
Comment (johnclary) [reply moped-database/migrations/1672774236734_create_table_public_moped_public_process_status/up.sql:1]: 💯 thank you @frankhereford for being a champion for consistency. @tillyw would you mind making this change in your branch?

File: moped-database/migrations/1672776463774_insert_public_process_statuses/up.sql
Comment (johnclary) [reply moped-database/migrations/1672776463774_insert_public_process_statuses/up.sql:6]: @tillyw one more minor change to the migrations — we no longer need this `Unknown` status since we're defaulting to `null`. would you remove it?

File: moped-database/migrations/1672774236734_create_table_public_moped_public_process_status/up.sql
Comment (mddilley) [reply moped-database/migrations/1672774236734_create_table_public_moped_public_process_status/up.sql:1]: +1 to this!

File: moped-editor/src/views/projects/projectView/ProjectSummary/ProjectSummary.js
Comment (mddilley) [inline moped-editor/src/views/projects/projectView/ProjectSummary/ProjectSummary.js:14]: My bad because I swept this out from under you. I removed `react-error-boundary` in this file and one other file in [this PR](https://github.com/cityofaustin/atd-moped/pull/915) since it wasn't a widely used pattern.

**Review by mddilley — APPROVED** [review]
Yay for reusable components! 🧱 🙌  Really awesome to see this change, and its immediate use to add another field. Can't wait to see it make another one easy breezy. Ready to 🚀 after that one update that John requested about the `Unknown` status.

File: moped-editor/src/views/projects/projectView/ProjectSummary/ProjectSummaryAutocomplete.js
Comment (tillyw) [reply moped-editor/src/views/projects/projectView/ProjectSummary/ProjectSummaryAutocomplete.js]: thanks @frankhereford! went with your suggestion

File: moped-database/migrations/1672774236734_create_table_public_moped_public_process_status/up.sql
Comment (tillyw) [reply moped-database/migrations/1672774236734_create_table_public_moped_public_process_status/up.sql]: agree with this table name pattern! if all the tables used this pattern we wouldn't need to feed so many props into this component, which would be nice. definitely something to stick with in the future!

Comment (tillyw) [reply moped-database/migrations/1672774236734_create_table_public_moped_public_process_status/up.sql]: thanks @frankhereford made those changes!

File: moped-database/migrations/1672776463774_insert_public_process_statuses/up.sql
Comment (tillyw) [reply moped-database/migrations/1672776463774_insert_public_process_statuses/up.sql]: done and done. thanks y'all!

**Comment by tillyw** [comment]
@frankhereford i believe i made all the migration changes as you requested. please let me know if i should rearrange anything?

File: moped-editor/src/views/projects/projectView/ProjectSummary/ProjectSummary.js
Comment (tillyw) [reply moped-editor/src/views/projects/projectView/ProjectSummary/ProjectSummary.js]: no worries, thanks for the heads up!

**Review by johnclary — CHANGES_REQUESTED** [review]
@tillyw this looks so good! thank you!!! this is ready to ship, but i'd like to hold off on merging until we can also get this field added to the activity log. i will open a separate issue for that.

**Comment by tillyw** [comment]
@johnclary gotcha - will do!

**Comment by tillyw** [comment]
merging - thanks for the reviews everyone!!

**Pull request merged**
